sdk/component/internal/storer/local/sqlite/sqlite_test.go

```diff
@@ -0,0 +1,218 @@
+				Count:           ptr(int32(1)),
+				Duration:        ptr(int32(3600)),
+				EndTime:         ptr(time.Now().Unix()),
+				EndTimeDt:       timestamppb.New(time.Now()),
+				Enrichments:     []*ocsf.Enrichment{{Type: ptr("Type1"), Value: "Value1"}},
+				Message:         ptr("Vulnerability finding message 1"),
+				Metadata: &ocsf.Metadata{
+					Version: "v1.0.1",
+				},
+				RawData:         ptr(`{"foo" : "bar"}`),
+				Severity:        ptr("Critical"),
+				SeverityId:      ocsf.VulnerabilityFinding_SEVERITY_ID_CRITICAL,
+				StartTime:       ptr(time.Now().Add(-time.Hour).Unix()),
+				StartTimeDt:     timestamppb.New(time.Now().Add(-time.Hour)),
+				Status:          ptr("Open"),
+				StatusCode:      ptr("200"),
+				Time:            time.Now().Unix(),
+				TimeDt:          timestamppb.New(time.Now()),
+				TimezoneOffset:  ptr(int32(-7)),
+				TypeName:        ptr("Type 1"),
+				TypeUid:         1,
+				Vulnerabilities: []*ocsf.Vulnerability{{Severity: ptr("Critical")}},
+			},
+			{
+				ActivityId:      ocsf.VulnerabilityFinding_ACTIVITY_ID_CREATE,
+				ActivityName:    ptr("Activity 2"),
+				CategoryName:    ptr("Category B"),
+				CategoryUid:     ocsf.VulnerabilityFinding_CATEGORY_UID_FINDINGS,
+				ClassName:       ptr("Class B"),
+				ClassUid:        ocsf.VulnerabilityFinding_CLASS_UID_VULNERABILITY_FINDING,
+				Cloud:           &ocsf.Cloud{Provider: "AWS", Region: ptr("us-east-2")},
+				Comment:         ptr("This is a comment for finding 2."),
+				Confidence:      ptr("High"),
+				ConfidenceId:    ptr(ocsf.VulnerabilityFinding_CONFIDENCE_ID_HIGH),
+				ConfidenceScore: ptr(int32(100)),
+				Count:           ptr(int32(5)),
+				Duration:        ptr(int32(3600)),
+				EndTime:         ptr(time.Now().Unix()),
+				EndTimeDt:       timestamppb.New(time.Now()),
+				Enrichments:     []*ocsf.Enrichment{{Type: ptr("Type2"), Value: "Value2"}},
+				Message:         ptr("Vulnerability finding message 2"),
+				Metadata: &ocsf.Metadata{
+					Version: "v1.0.1",
+				},
+				RawData:         ptr(`{"bar" : "baz"}`),
+				Severity:        ptr("Critical"),
+				SeverityId:      ocsf.VulnerabilityFinding_SEVERITY_ID_CRITICAL,
+				StartTime:       ptr(time.Now().Add(-time.Hour).Unix()),
+				StartTimeDt:     timestamppb.New(time.Now().Add(-time.Hour)),
+				Status:          ptr("Closed"),
+				StatusCode:      ptr("200"),
+				Time:            time.Now().Unix(),
+				TimeDt:          timestamppb.New(time.Now()),
+				TimezoneOffset:  ptr(int32(-7)),
+				TypeName:        ptr("Type 2"),
+				TypeUid:         2,
+				Vulnerabilities: []*ocsf.Vulnerability{{Severity: ptr("Critical")}},
+			},
+		}
+	)
+
+	defer cancel()
+
+	mts.t.Run("given an empty database, when I look for findings, I get none back", func(t *testing.T) {
+		resFindings, err := mts.manager.Read(
+			ctx,
+			workflowID,
+		)
+		require.Error(mts.t, err)
+		assert.ErrorIs(t, err, storer.ErrNoFindingsFound)
+		assert.Len(mts.t, resFindings, 0)
```

Comment:
why mix `assert` and `require` in these tests?

Reply:
Usually require should be used for something that must not fail to move on with further assertions. A require failure will halt the test case

Assertions can instead fail and the test case can keep going on

My rule of thumb is:
- use require to stop the world and don't have noisy tests - i.e. a panic caused by accessing a slice which is empty
- use assert when it's safe to keep the test case going - it's generally nice to use when you have multiple assertions so that you can immediately see all of them failing - with require this would lead to having to fix them one at a time to discover the issues - with assert it's only one run - this is very good on ci usually

Reply:
Yes but in this case this is adding verbosity without extra visibility. As in other places, this could be simplified to `require.ErrorIs(t, err, storer.ErrNoFindingsFound)` and then `require.Len(t, resFindings, 0)`. the last assertion is at the end of the test Run so it doesn't make a difference, but having a require will make the code more consistent